annotators/property_extraction/server.py

```diff
@@ -163,42 +169,45 @@ def get_result(request):
     logger.info(
         f"init_uttrs {init_uttrs} entities_with_labels: {entities_with_labels_batch} entity_info: {entity_info_batch}"
     )
-    uttrs = []
-    for uttr_list in init_uttrs:
-        if len(uttr_list) == 1:
-            uttrs.append(uttr_list[0].lower())
-        else:
-            utt_prev = uttr_list[-2]
-            utt_prev_sentences = nltk.sent_tokenize(utt_prev)
-            utt_prev = utt_prev_sentences[-1]
-            utt_cur = uttr_list[-1]
-            utt_prev_l = utt_prev.lower()
-            utt_cur_l = utt_cur.lower()
-            is_q = (
-                any([utt_prev_l.startswith(q_word) for q_word in ["what ", "who ", "when ", "where "]])
-                or "?" in utt_prev_l
-            )
-
-            is_sentence = False
-            parsed_sentence = nlp(utt_cur_l)
-            if parsed_sentence:
-                tokens = [elem.text for elem in parsed_sentence]
-                tags = [elem.tag_ for elem in parsed_sentence]
-                found_verbs = any([tag in tags for tag in ["VB", "VBZ", "VBP", "VBD"]])
-                if found_verbs and len(tokens) > 2:
-                    is_sentence = True
-
-            logger.info(f"is_q: {is_q} --- is_s: {is_sentence} --- utt_prev: {utt_prev_l} --- utt_cur: {utt_cur_l}")
-            if is_q and not is_sentence:
-                uttrs.append(sentrewrite(utt_prev_l, utt_cur_l))
+    if not init_uttrs[0][0]:
+        triplets_batch = [[""]]
+        uttrs = [""]
```

Comment:
that's a strange check for me. You jsut check the first element and assign uttrs to list of empty string. But you are not sure that there is only one element. And why do you do this here?
Probably it;'s better to iterate over the batch of input and check every sample and append to uttr empty string if the input init_uttrs[i] is empty (or whatever check is)

Reply:
@zucchini-nlp

Reply:
this was all used to manage empty or very long bot utterances. Now, I removed the check for empty and moved it for later when iterating the utterances. The `is_question`  function is also removed and I reverted the question-check to the way it was before.